app/src/main/java/edu/usf/sas/pal/muser/manager/UserRegistrationManager.java

```diff
@@ -0,0 +1,218 @@
+package edu.usf.sas.pal.muser.manager;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.support.v4.graphics.drawable.DrawableCompat;
+import android.text.Html;
+import android.text.TextUtils;
+import android.util.Log;
+import android.util.Patterns;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.simplecity.amp_library.R;
+import com.simplecity.amp_library.ShuttleApplication;
+import com.simplecity.amp_library.ui.screens.main.MainActivity;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import edu.usf.sas.pal.muser.constants.EventConstants;
+import edu.usf.sas.pal.muser.util.FirebaseIOUtils;
+import edu.usf.sas.pal.muser.util.PreferenceUtils;
+
+/**
+ * Manager class to handle user registration. Creates dialogs to prompt the user about the study
+ * fetch  email if user agrees to enroll.
+ */
+public class UserRegistrationManager {
+
+    private static String TAG = UserRegistrationManager.class.getSimpleName();
+
+    private Context mApplicationContext;
+
+    private Context mActivityContext;
+
+    public UserRegistrationManager(Context applicationContext, Context activityContext){
+        mActivityContext = activityContext;
+        mApplicationContext = activityContext;
+    }
+
+    public void registerParticipant(){
+        registerParticipant(false);
+    }
+
+    public void registerParticipant(boolean forceStart){
+        // TODO check possible limit of enrollment from a certain location?
+        boolean isUserOptOut = PreferenceUtils.getBoolean(EventConstants.USER_OPT_OUT,
+                false);
+        if (forceStart) isUserOptOut = false;
```

Comment:
Could you please wrap all IF statement bodies, even single lines, with `{` and `}`. I know this is a style thing but I prefer it this way.